core/src/main/scala/org/ensime/indexer/FileWatchers.scala

```diff
@@ -43,10 +46,33 @@ class ClassfileWatcher(
 
   private val impls =
     if (config.disableClassMonitoring) Nil
-    else config.targetClasspath.map { target =>
-      val (selector, dir, rec) =
-        if (target.isJar) (JarSelector, target.getParentFile, false) else (ClassfileSelector, target, true)
-      new ApachePollingFileWatcher(dir, selector, rec, listeners)
+    else {
+      def buildJava6Watcher() = {
+        config.targetClasspath.map { target =>
+          val (selector, dir, rec) =
+            if (target.isJar) (JarSelector, target.getParentFile, false) else (ClassfileSelector, target, true)
+          log.debug(s"creating a ApachePollingFileWatcher watcher for $dir")
+          new ApachePollingFileWatcher(dir, selector, rec, listeners).asInstanceOf[Watcher]
+        }
+      }
+      def buildJava7Watcher() = {
+        val jarJava7WatcherBuilder = new JarJava7WatcherBuilder()
+        val classJava7WatcherBuilder = new ClassJava7WatcherBuilder()
+        config.targetClasspath.map { target =>
+          if (target.isJar) {
+            log.debug(s"creating a Java 7 jar watcher for ${target}")
```

Comment:
I normally add log.ifDebugEnabled around these - somebody should write a macro that does that automatically, otherwise you are at a minimum creating a thunk for the log message.  (Nitpicking)